### PR TITLE
Add GNN training benchmark

### PR DESCRIPTION
**Pull request by drcanchi**

### CODE REVIEW

File: mlperf_logging/compliance_checker/training_4.0.0/closed_gnn.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+- KEY:
+    NAME:  global_batch_size
+    REQ:   EXACTLY_ONE
+    CHECK: " v['value'] > 0"
+
+- KEY:
+    NAME:  opt_name
```
Comment (Elnifio) [inline mlperf_logging/compliance_checker/training_4.0.0/closed_gnn.yaml:7]: In the reference code, the optimizer name is not reported, but we do have a `SEED` exported at [this line](https://github.com/mlcommons/training/blob/3afa89d9094e975aa77b3e1e2af41a2e0af9b0ed/graph_neural_network/train_rgnn_multi_gpu.py#L317). Should we report the optimizer name as well in the reference code? And should we also check if the seed is included in the log as well?

File: mlperf_logging/compliance_checker/training_4.0.0/closed_common.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+- KEY:
+    NAME:  submission_benchmark
+    REQ:   EXACTLY_ONE
+    CHECK: " v['value'] in ['resnet', 'ssd', 'stable_diffusion', 'maskrcnn', 'gpt3', 'dlrm_dcnv2', 'bert', 'rnnt', 'unet3d', 'gnn'] "
+    POST:  " enqueue_config('training_4.0.0/closed_{}.yaml'.format(v['value'])) "
+
+- KEY:
+    NAME: gradient_accumulation_steps
```
Comment (Elnifio) [inline mlperf_logging/compliance_checker/training_4.0.0/closed_common.yaml:9]: We actually do not have this key in the reference implementation (`mllog.constants.GRADIENT_ACCUMULATION_STEPS`).  Should we also include this in the reference branch? Or is it okay for us to directly remove this in the `closed_common.yaml` checker?

File: mlperf_logging/compliance_checker/training_4.0.0/closed_gnn.yaml (hunk 1 above)
Comment (drcanchi) [reply mlperf_logging/compliance_checker/training_4.0.0/closed_gnn.yaml:7]: We should report optimizer name in reference. The seed check is part of package_checker and we don't need to include it here.

File: mlperf_logging/compliance_checker/training_4.0.0/closed_common.yaml (hunk 2 above)
Comment (drcanchi) [reply mlperf_logging/compliance_checker/training_4.0.0/closed_common.yaml:9]: Let's include gradient acccumulation in reference, as we should not be modifying the common yaml.

File: mlperf_logging/compliance_checker/training_4.0.0/closed_gnn.yaml (hunk 1 above)
Comment (LiSu) [reply mlperf_logging/compliance_checker/training_4.0.0/closed_gnn.yaml:7]: Added optimizer name in reference.

File: mlperf_logging/compliance_checker/training_4.0.0/closed_common.yaml (hunk 2 above)
Comment (LiSu) [reply mlperf_logging/compliance_checker/training_4.0.0/closed_common.yaml:9]: Gradient accumulation is included in reference.

**Comment by mrmhodak** [comment]
LGTM

**Comment by nv-rborkar** [comment]
@drcanchi please resolve the conflicts so that we can merge. Thanks!

**Review by hiwotadese — APPROVED** [review]
